From 54bfabc270b1185f58bc1af03eaa320d3fb4ecd1 Mon Sep 17 00:00:00 2001
Message-Id: <54bfabc270b1185f58bc1af03eaa320d3fb4ecd1.1268128506.git.tanguy.ortolo@neolane.com>
From: Tanguy Ortolo <tanguy+debian@ortolo.eu>
Date: Mon, 8 Mar 2010 23:06:21 +0100
Subject: [PATCH] Add Xsession script to work around the Java bug
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="------------1.6.5.3"

This is a multi-part message in MIME format.
--------------1.6.5.3
Content-Type: text/plain; charset=UTF-8; format=fixed
Content-Transfer-Encoding: 8bit


This script sets _JAVA_AWT_WM_NONREPARENTING environment variable
to expose awesome as a non-reparenting window manager, as explained
in the Debian bug #508650.
---
 debian/55awesome-javaworkaround |   10 ++++++++++
 debian/awesome.install          |    1 +
 2 files changed, 11 insertions(+), 0 deletions(-)
 create mode 100644 debian/55awesome-javaworkaround


--------------1.6.5.3
Content-Type: text/x-patch; name="0001-Add-Xsession-script-to-work-around-the-Java-bug.patch"
Content-Transfer-Encoding: 8bit
Content-Disposition: attachment; filename="0001-Add-Xsession-script-to-work-around-the-Java-bug.patch"

diff --git a/debian/55awesome-javaworkaround b/debian/55awesome-javaworkaround
new file mode 100644
index 0000000..c7b8d62
--- /dev/null
+++ b/debian/55awesome-javaworkaround
@@ -0,0 +1,10 @@
+# If we are running the awesome, expose it to Java as a non-reparenting window
+# manager. Cf. bug #508650.
+
+BASESTARTUP=$(basename "$STARTUP" | cut -d\  -f1)
+if [ "$BASESTARTUP" = x-session-manager ]; then
+    BASESTARTUP=$(basename $(readlink /etc/alternatives/x-session-manager))
+fi
+if [ "$BASESTARTUP" = "awesome" ]; then
+    _JAVA_AWT_WM_NONREPARENTING=1; export _JAVA_AWT_WM_NONREPARENTING
+fi
diff --git a/debian/awesome.install b/debian/awesome.install
index e0b17b0..e50e2c7 100644
--- a/debian/awesome.install
+++ b/debian/awesome.install
@@ -1 +1,2 @@
 debian/awesome.xpm usr/share/pixmaps
+debian/55awesome-javaworkaround /etc/X11/Xsession.d

--------------1.6.5.3--


